### DISCUSSION TITLE
Question about the RNA-seq data

### DISCUSSION THREAD

**Discussion by SimonLi5601** [broadinstitute/seqr#2234] [Q&A]
Hi team,  we noticed recently you added the RNA-seq feature to the application, congratulations! It's awesome! We are interested to bring this feature to our in-house version as well. It seems this feature requires RNA-seq coverage and junction file. We were wondering what tools are you using to prepare those files. If you could provides some additional information, that will be great! Thanks a lot in advance!

**Comment by hanars** [comment]
Hi Simon! @bw2 has been doing the data generation for the junction/ coverage files - hopefully he can give you more information about how thats done

**Reply by SimonLi5601** [reply]
Thanks, Hana! Hi, @bw2 could you share with us what tools you were using for the RNA-seq junction/coverage files? It would be very helpful.

**Reply by bw2** [reply]
Hi @SimonLi5601, yes, the RNA-seq track is based on a .bigWig file + a .bed file that contains a row for each splice junction arc.

The .bed file format is:
```
chr1	11211	12009	motif=GT/AG;uniquely_mapped=0;multi_mapped=1;maximum_spliced_alignment_overhang=12;annotated_junction=false;num_samples_with_this_junction=16;num_samples_total=243	0	+
chr1	11671	12009	motif=GT/AG;uniquely_mapped=0;multi_mapped=4;maximum_spliced_alignment_overhang=27;annotated_junction=false;num_samples_with_this_junction=56;num_samples_total=243	0	+
chr1	12227	12612	motif=GT/AG;uniquely_mapped=0;multi_mapped=3;maximum_spliced_alignment_overhang=7;annotated_junction=true;num_samples_with_this_junction=27;num_samples_total=243	0	+
```
and you can find additional docs @ https://github.com/igvteam/igv.js/wiki/Splice-Junctions

You can generate this .bed file by running https://github.com/broadinstitute/tgg-viewer/blob/master/sample_preprocessing/splice_junction_track/convert_SJ_out_tab_to_junctions_bed.py
on each sample's STAR splice junction output table (*.SJ.out.tab) optionally also passing in a Gencode gff file to annotate known vs. novel junctions.

To generate the coverage bigWig, you can use 
https://github.com/broadinstitute/tgg-viewer/blob/master/sample_preprocessing/splice_junction_track/wdl/generate-coverage-bigWig.wdl

or use some other tool like https://deeptools.readthedocs.io/en/develop/content/tools/bamCoverage.html

If you're storing your output files in google buckets, you can also use https://tgg-viewer.broadinstitute.org/ to test out your tracks by logging in with a google account that has access to your bucket, and then adding the data paths using 
![image](https://user-images.githubusercontent.com/6240170/139154378-72ecc9a4-29d0-4244-bfb1-ee3d5e955211.png)

**Reply by SimonLi5601** [reply]
Thanks a lot, @bw2. It's very helpful! I will look into it.

**Reply by zhanglw05** [reply]
@bw2 , we are working on adding RNAseq  feature to our in-house version. Two questions have been brought up by our collaborators. 1. Do you do any sort of normalization of your samples to keep them consistent with data previously loaded in the system (for junction counts and coverage); 2. have you considered normalizing the GTEx coverage data so users can compare expression in their sample to GTEx based on depth of coverage.
